ui/app/components/wizard-content.js

```diff
@@ -1,13 +1,100 @@
 import { inject as service } from '@ember/service';
 import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { FEATURE_MACHINE_STEPS, INIT_STEPS } from 'vault/helpers/wizard-constants';
+
 export default Component.extend({
   wizard: service(),
   classNames: ['ui-wizard'],
   glyph: null,
   headerText: null,
+  selectProgress: null,
+  currentMachine: computed.alias('wizard.currentMachine'),
+  tutorialState: computed.alias('wizard.currentState'),
+  tutorialComponent: computed.alias('wizard.tutorialComponent'),
+  showProgress: computed('wizard.featureComponent', 'tutorialState', function() {
+    return (
+      this.tutorialComponent.includes('active') &&
+      (this.tutorialState.includes('init.active') ||
+        (this.wizard.featureComponent && this.wizard.featureMachineHistory))
+    );
+  }),
+  featureMachineHistory: computed.alias('wizard.featureMachineHistory'),
+  totalFeatures: computed('wizard.featureList', function() {
+    return this.wizard.featureList.length;
+  }),
+  completedFeatures: computed('wizard.currentMachine', function() {
+    return this.wizard.getCompletedFeatures();
+  }),
+  currentFeatureProgress: computed('featureMachineHistory.[]', function() {
+    if (this.tutorialState.includes('active.feature')) {
+      let totalSteps = FEATURE_MACHINE_STEPS[this.currentMachine];
+      if (this.currentMachine === 'secrets') {
```

Comment:
I haven't thought through this so could be a terrible idea - but is there a way to push this stuff into the machine somewhere? Then we could just get the state node for the feature from the machine and maybe there'd be a totalSteps count there?

Reply:
I think we'd have to add it to _every_ node. and some nodes (like display) are used for multiple states so it'd be hard

Reply:
Eesh yeah that sounds not fun.